src/mono/wasm/runtime/es6/dotnet.es6.lib.js

```diff
@@ -73,115 +75,36 @@ const postset = `
-let linked_functions = [
-    // mini-wasm.c
-    "mono_wasm_schedule_timer",
-
-    // mini-wasm-debugger.c
-    "mono_wasm_asm_loaded",
-    "mono_wasm_fire_debugger_agent_message_with_data",
-    "mono_wasm_debugger_log",
-    "mono_wasm_add_dbg_command_received",
-    "mono_wasm_set_entrypoint_breakpoint",
-
-    // mono-threads-wasm.c
-    "schedule_background_exec",
-
-    // interp.c
-    "mono_wasm_profiler_enter",
-    "mono_wasm_profiler_leave",
-
-    // driver.c
-    "mono_wasm_trace_logger",
-    "mono_wasm_event_pipe_early_startup_callback",
-
-    // jiterpreter.c / interp.c / transform.c
-    "mono_interp_tier_prepare_jiterpreter",
-    "mono_interp_record_interp_entry",
-    "mono_interp_jit_wasm_entry_trampoline",
-    "mono_interp_jit_wasm_jit_call_trampoline",
-    "mono_interp_invoke_wasm_jit_call_trampoline",
-    "mono_interp_flush_jitcall_queue",
-    "mono_jiterp_do_jit_call_indirect",
-
-    // corebindings.c
-    "mono_wasm_release_cs_owned_object",
-    "mono_wasm_bind_js_function",
-    "mono_wasm_invoke_bound_function",
-    "mono_wasm_invoke_import",
-    "mono_wasm_bind_cs_function",
-    "mono_wasm_marshal_promise",
-    "mono_wasm_change_case_invariant",
-    "mono_wasm_change_case",
-    "mono_wasm_compare_string",
-    "mono_wasm_starts_with",
-    "mono_wasm_ends_with",
-    "mono_wasm_index_of",
-    "mono_wasm_get_calendar_info",
-    "mono_wasm_get_culture_info",
-    "mono_wasm_get_first_day_of_week",
-    "mono_wasm_get_first_week_of_year",
-
-    "icudt68_dat",
-];
+// the JS methods would be visible to EMCC linker and become imports of the WASM module
+// we generate simple stub for each exported function so that emcc will include them in the final output
+// we will replace them with the real implementation in replace_linker_placeholders
+function createWasmImportStubs() {
+    for (let functionName in methodIndexByName.mono_wasm_imports) {
+        const idx = methodIndexByName.mono_wasm_imports[functionName];
+        DotnetSupportLib[functionName] = new Function(`return {runtime_idx:${idx}};//${functionName}`);
+    }
 
-#if USE_PTHREADS
-linked_functions = [...linked_functions,
-    // mono-threads-wasm.c
-    "mono_wasm_pthread_on_pthread_attached",
-    "mono_wasm_pthread_on_pthread_detached",
-    // threads.c
-    "mono_wasm_eventloop_has_unsettled_interop_promises",
-    // diagnostics_server.c
-    "mono_wasm_diagnostic_server_on_server_thread_created",
-    "mono_wasm_diagnostic_server_on_runtime_server_init",
-    "mono_wasm_diagnostic_server_stream_signal_work_available",
-    // corebindings.c
-    "mono_wasm_install_js_worker_interop",
-    "mono_wasm_uninstall_js_worker_interop",
-]
-#endif
+    #if USE_PTHREADS
+    for (let functionName in methodIndexByName.mono_wasm_threads_imports) {
+        const idx = methodIndexByName.mono_wasm_threads_imports[functionName];
+        DotnetSupportLib[functionName] = new Function(`return {runtime_idx:${idx}};//${functionName}`);
```

Comment:
As I mentioned in the approval comment, since we're doing x[y] here instead of the equivalent of Dictionary.Add, a name collision will silently break things

Reply:
added assert and refactored the logic